File: Easy/1786-count-the-number-of-consistent-strings/count-the-number-of-consistent-strings.py
class Solution:
    def countConsistentStrings(self, allowed: str, words: List[str]) -> int:
        count = 0
        allowed = set(allowed)
        for word in words:
            for w in word:
                if w not in allowed:
                    count += 1
                    break
        return len(words) - count

File: 1786-count-the-number-of-consistent-strings/count-the-number-of-consistent-strings.cpp
#include <iostream>
#include <vector>
#include <unordered_set>
using namespace std;

class Solution {
public:
    int countConsistentStrings(string allowed, vector<string>& words) {
        int count = 0;
        unordered_set<char> allowedSet(allowed.begin(), allowed.end());

        for (const auto& word : words) {
            bool consistent = true;
            for (char w : word) {
                if (allowedSet.find(w) == allowedSet.end()) {
                    consistent = false;
                    break;
                }
            }
            if (consistent) {
                count++;
            }
        }

        return count;
    }
};